lib/rules/padding-line-between-statements.js

```diff
@@ -537,40 +563,46 @@ module.exports = {
                     match(nextNode, configure.next);
 
                 if (matched) {
-                    return PaddingTypes[configure.blankLine];
+                    if (typeof configure.blankLine === "number") {
+                        return [configure.blankLine, PaddingTypes.always];
+                    }
+                    return [0, PaddingTypes[configure.blankLine]];
                 }
             }
-            return PaddingTypes.any;
+            return [0, PaddingTypes.any];
         }
 
         /**
          * Gets padding line sequences between the given 2 statements.
          * Comments are separators of the padding line sequences.
          * @param {ASTNode} prevNode The previous statement to count.
          * @param {ASTNode} nextNode The current statement to count.
+         * @param {number} blankLine Allowed number of blank lines.
          * @returns {Array<Token[]>} The array of token pairs.
          * @private
          */
-        function getPaddingLineSequences(prevNode, nextNode) {
+        function getPaddingLineSequences(prevNode, nextNode, blankLine) {
             const pairs = [];
             let prevToken = getActualLastToken(sourceCode, prevNode);
+            let foundBlankLine = 0;
 
-            if (nextNode.loc.start.line - prevToken.loc.end.line >= 2) {
-                do {
-                    const token = sourceCode.getTokenAfter(
-                        prevToken,
-                        { includeComments: true }
-                    );
+            while (prevToken.range[0] < nextNode.range[0]) {
+                const token = sourceCode.getTokenAfter(
+                    prevToken,
+                    { includeComments: true }
+                );
+                const currentBlankLine = token.loc.start.line - prevToken.loc.end.line - 1;
 
-                    if (token.loc.start.line - prevToken.loc.end.line >= 2) {
-                        pairs.push([prevToken, token]);
-                    }
-                    prevToken = token;
-
-                } while (prevToken.range[0] < nextNode.range[0]);
+                if (currentBlankLine > foundBlankLine) {
+                    foundBlankLine = currentBlankLine;
+                }
+                if (blankLine ? (currentBlankLine === blankLine) : (currentBlankLine >= 1)) {
+                    pairs.push([prevToken, token]);
+                }
+                prevToken = token;
             }
 
-            return pairs;
+            return [foundBlankLine, pairs];
```

Comment:
This changes the return type signature of this function. Can we update the JSDoc block?